src/snowflake/snowpark/dataframe_reader.py

```diff
@@ -452,3 +471,33 @@ def _read_semi_structured_file(self, path: str, format: str) -> DataFrame:
         )
         df._reader = self
         return df
+
+    def _upload_local_file_to_stage(self, path: str, format: str) -> str:
```

Comment:
I'm thinking if the following flow makes sense:

```
if path does not exist:
    do we want to do something?
filelist = []
if path is file:
    filelist.append(path)
    stage_path = stage join path
elif path is directory:
    filelist = listdir(path)
    stage_path = stage

for file in filelist:
    do path transformation and put

return stage_path
```

Reply:
ok, i will add a valueerror exception if the path does not exist

Reply:
I think we do ignore and do nothing if the path does not exist because when you want to access a file on stage, whose path does not exist in local machine, you dont need to do anything else then the previous code do

Reply:
hmm I think it'd better to raise a ValueError here if the local file doesn't exist. Otherwise the error message the user will get is misleading saying stage file doesn't exist, but actually, the local file doesn't exist. 

You can just check whether it's remote file by:
```
from snowflake.snowpark._internal.utils import STAGE_PREFIX
if path.startswith(STAGE_PREFIX):
    ...
```

Also, you should add a negative test to test when
1) the input is a local file and the file doesn't exist
2) the input is a local directory, and we want to read parquet files.